maro/data_lib/cim/vessel_future_stops_prediction.py

```diff
@@ -37,41 +39,40 @@ def __getitem__(self, key):
         # ignore current port if parking
         last_stop_idx = loc_idx + (0 if last_loc_idx == loc_idx else -1)
 
-        last_stop = self._stops[vessel_idx][last_stop_idx]
-        last_port_idx = last_stop.port_idx
-        last_port_arrival_tick = last_stop.arrival_tick
+        return self._predict_future_stops(vessel_idx, last_stop_idx, self._stop_number)
 
-        return self._predict_future_stops(vessel_idx, last_port_idx, last_port_arrival_tick, self._stop_number)
+    def _make_vessel_start_port_offset(self) -> List[int]:
+        vessel_start_port_offsets = []
+        for vessel in self._vessels:
+            route_points = self._routes[self._route_mapping[vessel.route_name]]
+            route_point_names = [rp.port_name for rp in route_points]
+            vessel_start_port_offset = route_point_names.index(vessel.start_port_name)
+            vessel_start_port_offsets.append(vessel_start_port_offset)
```

Comment:
The `vessel_start_port_offsets` is initialized according to **the order** in `self._vessels`, but is used by indexing with the `vessel_idx`. I just think it heavily depends on the implementation of the frame builder. Should we change it to a `dict`?